libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -503,9 +503,15 @@ void PbftManager::initialState_() {
 
   auto soft_voted_block_hash = db_->getPbftMgrVotedValue(PbftMgrVotedValue::SoftVotedBlockHashInRound);
   auto soft_voted_block = db_->getPbftMgrStatus(PbftMgrStatus::SoftVotedBlockInRound);
+
   if (soft_voted_block_hash) {
     // From DB
     soft_voted_block_for_this_round_ = std::make_pair(*soft_voted_block_hash, soft_voted_block);
+    for (const auto &vote : db_->getSoftVotes(round)) {
+      if (vote->getBlockHash() == *soft_voted_block_hash) {
+        vote_mgr_->addVerifiedVote(vote);
```

Comment:
so as your are basically only removing functionality that saved votes into the db, how is possible that this is needed and wasnt before ?

Reply:
yeah, so this is there in case of crash after soft votes are saved, before we had them in DB as all votes where saved, so no need to recover them.